File: lib/sponsorship.js
const NONE = [
  /\bdo(?:es)? not sponsor\b/i,
  /\bno (?:visa )?sponsorship\b/i,
  /\bunable to (?:provide|offer) (?:visa )?sponsorship\b/i,
  /without (?:visa )?sponsorship\b/i,
  /must be authorized to work in the u\.?s/i,
  /u\.?s\.? citizens? only/i,
  /security clearance/i,
  /not (?:able|eligible) to sponsor\b/i,
];

const AVAILABLE = [
  /\b(?:visa )?sponsorship (?:is )?available\b/i,
  /\bwill sponsor\b/i,
  /\bwe sponsor\b/i,
  /\bh-?1b sponsorship\b/i,
  /sponsorship (?:is )?(?:provided|offered)\b/i,
  /relocation and visa (?:support )?(?:provided|available)?\b/i,
];

function sponsorshipSignal(text) {
  const t = String(text || "");
  if (!t.trim()) return "unknown";
  if (NONE.some((r) => r.test(t))) return "none";
  if (AVAILABLE.some((r) => r.test(t))) return "available";
  return "unknown";
}

module.exports = { sponsorshipSignal };

File: crawler/sponsorship.py
import re
from typing import Optional


NONE = [
    re.compile(r"\bdo(?:es)? not sponsor\b", re.I),
    re.compile(r"\bno (?:visa )?sponsorship\b", re.I),
    re.compile(r"\bunable to (?:provide|offer) (?:visa )?sponsorship\b", re.I),
    re.compile(r"without (?:visa )?sponsorship\b", re.I),
    re.compile(r"must be authorized to work in the u\.?s", re.I),
    re.compile(r"u\.?s\.? citizens? only", re.I),
    re.compile(r"security clearance", re.I),
    re.compile(r"not (?:able|eligible) to sponsor\b", re.I),
]

AVAILABLE = [
    re.compile(r"\b(?:visa )?sponsorship (?:is )?available\b", re.I),
    re.compile(r"\bwill sponsor\b", re.I),
    re.compile(r"\bwe sponsor\b", re.I),
    re.compile(r"\bh-?1b sponsorship\b", re.I),
    re.compile(r"sponsorship (?:is )?(?:provided|offered)\b", re.I),
    re.compile(r"relocation and visa (?:support )?(?:provided|available)?\b", re.I),
]


def sponsorship_signal(text: Optional[str]) -> str:
    t = str(text or "")
    if not t.strip():
        return "unknown"
    if any(r.search(t) for r in NONE):
        return "none"
    if any(r.search(t) for r in AVAILABLE):
        return "available"
    return "unknown"
